File: tests/test_php/src/Services/OrderService.php
<?php

namespace App\Services;

use App\Models\Order;
use App\Models\User;
use DateTime;

class OrderService {
    public function process(Order $order, User $user = null) {
        echo "Processing order " . $order->getId() . "\n";
        if ($user) {
            echo "For user: " . $user->getName() . "\n";
        }
        echo "Total: " . $order->formatTotal() . "\n";
        return true;
    }

    public function cancel($orderId) {
        $now = new DateTime();
        echo "Canceling order $orderId at " . $now->format('Y-m-d H:i:s') . "\n";
        return true;
    }

    public function getOrdersByUser(User $user) {
        return array();
    }
}
?>

File: tests/test_zig/services/order_service.zig
const std = @import("std");
const Order = @import("../models/order.zig");
const User = @import("../models/user.zig");

pub const OrderService = struct {
    pub fn init() OrderService {
        return OrderService{};
    }

    pub fn process(self: *OrderService, order: *const Order.Order) void {
        _ = self;
        std.debug.print("Processing order: {d}\n", .{order.id});
    }
};
